internal/appfunction/context.go

```diff
@@ -280,44 +282,41 @@ func (appContext *Context) PipelineId() string {
 }
 
 // Publish pushes data to the MessageBus using configured topic
-func (appContext *Context) Publish(data any) error {
-	messageClient := bootstrapContainer.MessagingClientFrom(appContext.Dic.Get)
-	if messageClient == nil {
-		return fmt.Errorf(messageBusDisabledErr)
-	}
+func (appContext *Context) Publish(data any, contentType string) error {
+	topic := container.ConfigurationFrom(appContext.Dic.Get).Trigger.PublishTopic
 
-	triggerConfig := container.ConfigurationFrom(appContext.Dic.Get).Trigger
-	baseTopic := bootstrapContainer.ConfigurationFrom(appContext.Dic.Get).GetBootstrap().MessageBus.BaseTopicPrefix
-
-	payload, err := json.Marshal(data)
-	if err != nil {
-		return fmt.Errorf("failed to marshal data for publishing using configured topic: %v", err)
-	}
-	message := types.NewMessageEnvelope(payload, context.Background())
-	err = messageClient.Publish(message, common.BuildTopic(baseTopic, triggerConfig.PublishTopic))
+	err := appContext.PublishWithTopic(topic, data, contentType)
 	if err != nil {
-		return fmt.Errorf("%v: %v", publishDataErr, err)
```

Comment:
why not return `fmt.Errorf("%v: %v", publishDataErr, err)`

Reply:
Because PublishWithTopic already does this

Reply:
I see now that it is not needed, thank you for the clarification